adafruit_tmp007.py

```diff
@@ -0,0 +1,197 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_tmp007`
+====================================================
+
+CircuitPython driver for the TMP007 contactless IR thermometer
+
+* Author(s): Jerry Needell
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+"""
+
+# imports
+
+import time
+from micropython import const
+from adafruit_bus_device.i2c_device import I2CDevice
+
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_TMP007.git"
+
+
+# Default device I2C address.
+_TMP007_I2CADDR = const(0x40)
+
+# Register addresses.
+_TMP007_CONFIG = const(0x02)
+_TMP007_DEVID = const(0x1F)
+_TMP007_VOBJ = const(0x0)
+_TMP007_TAMB = const(0x01)
+_TMP007_TOBJ = const(0x03)
+
+# Config register values.
+_TMP007_CFG_RESET = const(0x8000)
+_TMP007_CFG_MODEON = const(0x7000)
+CFG_1SAMPLE = const(0x0000)
+CFG_2SAMPLE = const(0x0200)
+CFG_4SAMPLE = const(0x0400)
+CFG_8SAMPLE = const(0x0600)
+CFG_16SAMPLE = const(0x0800)
+_TMP007_CFG_DRDYEN = const(0x0100)
+_TMP007_CFG_DRDY = const(0x0080)
+
+
+class TMP007:
+    """Class to represent an Adafruit TMP007 non-contact temperature measurement
+    board.
+    """
+
+    # Class-level buffer for reading and writing data with the sensor.
+    # This reduces memory allocations but means the code is not re-entrant or
+    # thread safe!
+    _BUFFER = bytearray(4)
+
+
+
+    def __init__(self, i2c, address=_TMP007_I2CADDR):
+        """Initialize TMP007 device on the specified I2C address and bus number.
+        Address defaults to 0x40 and bus number defaults to the appropriate bus
+        for the hardware.
+        """
+        self._device = I2CDevice(i2c, address)
+
+    def begin(self, samplerate=CFG_16SAMPLE):
```

Comment:
Fold this into `__init__` so its going immediately. It can always be slept after.

Reply:
I  merged begin int init and just throw an error if the ID is not valid. Is this what you wanted?